pkg/controller/storagecluster/reconcile.go

```diff
@@ -1152,3 +1156,129 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+//ensureJobTemplates returns spec of a job template
+func (r *ReconcileStorageCluster) ensureJobTemplates(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	osdCleanUpTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name + "-osd-removal",
+			Namespace: sc.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, osdCleanUpTemplate, func() error {
+		osdCleanUpTemplate.Objects = []runtime.RawExtension{
+			{
+				Object: newCleanupJob(sc),
+			},
+		}
+		osdCleanUpTemplate.Parameters = []openshiftv1.Parameter{
+			{
+				Name:     "ROOK_IMAGE",
+				Required: true,
+			},
+			{
+				Name:     "FAILED_OSD_ID",
+				Required: true,
+			},
+		}
```

Comment:
Can we move this to part where you initialize `osdCleanUpTemplate` ?
This doesn't look like it needs to be separate.

Reply:
Well, this is a mutating function... my guess would be that the fetched object gets injected into `osdCleanupTemplate`, so then we have to modify whatever is set to make sure it is reconciled properly. @xenolinux is that correct?

Reply:
Yes. That's right

Reply:
yup can be done